packages/framework/fluid-framework/src/index.ts

```diff
@@ -19,31 +19,56 @@ export { ConnectionState } from "@fluidframework/container-loader";
 export type {
 	ContainerAttachProps,
 	ContainerSchema,
-	DataObjectClass,
 	IConnection,
 	IFluidContainer,
 	IFluidContainerEvents,
 	IMember,
 	InitialObjects,
 	IServiceAudience,
 	IServiceAudienceEvents,
-	LoadableObjectClass,
-	LoadableObjectClassRecord,
 	MemberChangedListener,
 	Myself,
 } from "@fluidframework/fluid-static";
+export type { SharedObjectKind, ISharedObjectEvents } from "@fluidframework/shared-object-base";
 
 // Let the tree package manage its own API surface, we will simply reflect it here.
 // Note: this only surfaces the `@public` API items from the tree package. If the `@beta` and `@alpha` items are
 // desired, they can be added by re-exporting from one of the package's aliased export paths instead (e.g. `tree
 // alpha` to surface everything `@alpha` and higher).
-// eslint-disable-next-line no-restricted-syntax
+// eslint-disable-next-line no-restricted-syntax, import/export
 export * from "@fluidframework/tree";
 
+import type { SharedObjectKind } from "@fluidframework/shared-object-base";
+import { SharedTree as OriginalSharedTree } from "@fluidframework/tree/internal";
+import { type ITree } from "@fluidframework/tree";
+
+/**
+ * A hierarchical data structure for collaboratively editing strongly typed JSON-like trees
+ * of objects, arrays, and other data types.
+ * @public
+ */
+// eslint-disable-next-line import/export
+export const SharedTree: SharedObjectKind<ITree> = OriginalSharedTree;
+
 // ===============================================================
 // Legacy exports
 
-export { ContainerErrorTypes } from "@fluidframework/container-definitions/internal";
+export {
+	ContainerErrorTypes,
+	type IDeltaManager,
+	type IDeltaManagerEvents,
+	type IDeltaSender,
+	type IDeltaQueue,
+	type ReadOnlyInfo,
+	type IConnectionDetails,
+	type IDeltaQueueEvents,
+} from "@fluidframework/container-definitions/internal";
```

Comment:
Sorry if it's stupid question (have not looked at fluid-framework), but does this reexport essentially makes APIs "public"? I hope the answer is no, but worth checking.

Reply:
re-exports have no impact on the tagging. These types, which were transitively reachable from existing legacy/alpha API are simply being added to the legacy exports here, but they were already available under legacy on other packages. They were added because API extractor is configured to force it with the current rather indecisive config which forces including some transitive deps but not others (and unrelated changes in this PR impacted which transitive's it requires by impacting which types and packages are referenced directly)

Reply:
I think its a topic of some debate current if fluid-framework/legacy counts as "public" or more public than legacy on other packages which is why I was all wordy and specific there.